src/Toggle/Segment/SegmentPolicyEvaluator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Clearbooks\Labs\Client\Toggle\Segment;
+
+use Clearbooks\Labs\Client\Toggle\Gateway\SegmentTogglePolicyGateway;
+
+class SegmentPolicyEvaluator
+{
+    /**
+     * @var SegmentPriorityArranger
+     */
+    private $segmentPriorityArranger;
+
+    /**
+     * @var SegmentTogglePolicyGateway
+     */
+    private $segmentTogglePolicyGateway;
+
+    /**
+     * @param SegmentPriorityArranger $segmentPriorityArranger
+     * @param SegmentTogglePolicyGateway $segmentTogglePolicyGateway
+     */
+    public function __construct( SegmentPriorityArranger $segmentPriorityArranger,
+                                 SegmentTogglePolicyGateway $segmentTogglePolicyGateway )
+    {
+
+        $this->segmentPriorityArranger = $segmentPriorityArranger;
+        $this->segmentTogglePolicyGateway = $segmentTogglePolicyGateway;
+    }
+
+    /**
+     * @param array $segments
+     * @param string $toggleName
+     * @return bool|null
+     */
+    public function evaluateSegmentPoliciesForToggle( $toggleName, array $segments )
```

Comment:
It evaluates segment policies:
* Segment policies can enable or disable a toggle for a segment.
* User - group pairs can be part of multiple segments at the same time, or even none.
* If the current user - group pair doesn't have any segments we return null.
* If the current user - group pair is in some segments, but there's no toggle policy set for any of those segments, we once again return null.
* There's a priority order between different kind of policies. Null means here that the segment policies won't affect the toggle's active status, it will be decided by something else.